multiple-value uuid.NewV4() in single-value context

```
➜  go install github.com/koding/kite
# github.com/koding/kite
$GOPATH/src/github.com/koding/kite/kite.go:174:22: multiple-value uuid.NewV4() in single-value context
```

--- a/Gopkg.lock
+++ b/Gopkg.lock
@@ -27,15 +27,9 @@
 
 [[projects]]
   name = "github.com/coreos/etcd"
-  packages = [
-    "client",
-    "pkg/pathutil",
-    "pkg/srv",
-    "pkg/types",
-    "version"
-  ]
-  revision = "d3c2acf09011e89d61a8b3ec0f844411e200abb5"
-  version = "v3.3.0-rc.1"
+  packages = ["client","pkg/pathutil","pkg/srv","pkg/types","version"]
+  revision = "33245c6b5b49130ca99280408fadfab01aac0e48"
+  version = "v3.3.8"
 
 [[projects]]
   name = "github.com/coreos/go-semver"
@@ -46,8 +40,8 @@
 [[projects]]
   name = "github.com/dgrijalva/jwt-go"
   packages = ["."]
-  revision = "dbeaa9332f19a944acb5736b4456cfcc02140e29"
-  version = "v3.1.0"
+  revision = "06ea1031745cb8b3dab3f6a236daf2b0aa468b7e"
+  version = "v3.2.0"
 
 [[projects]]
   branch = "master"
@@ -58,8 +52,8 @@
 [[projects]]
   name = "github.com/fatih/color"
   packages = ["."]
-  revision = "570b54cabe6b8eb0bc2dfce68d964677d63b5260"
-  version = "v1.5.0"
+  revision = "5b77d2a35fb0ede96d138fc9a99f5c9b6aef11b4"
+  version = "v1.7.0"
 
 [[projects]]
   name = "github.com/fatih/structs"
@@ -70,14 +64,14 @@
 [[projects]]
   name = "github.com/gorilla/context"
   packages = ["."]
-  revision = "1ea25387ff6f684839d82767c1733ff4d4d15d0a"
-  version = "v1.1"
+  revision = "08b5f424b9271eedf6f9f0ce86cb9396ed337a42"
+  version = "v1.1.1"
 
 [[projects]]
   name = "github.com/gorilla/mux"
   packages = ["."]
-  revision = "7f08801859139f86dfafd1c296e2cba9a80d292e"
-  version = "v1.6.0"
+  revision = "e3702bed27f0d39777b0b37b664b6280e8ef8fbf"
+  version = "v1.6.2"
 
 [[projects]]
   name = "github.com/gorilla/websocket"
@@ -101,7 +95,7 @@
   branch = "master"
   name = "github.com/hashicorp/go-version"
   packages = ["."]
-  revision = "4fe82ae3040f80a03d04d2cccb5606a626b8e1ee"
+  revision = "23480c0665776210b5fbbac6eaaee40e3e6a96b7"
 
 [[projects]]
   name = "github.com/igm/sockjs-go"
@@ -109,10 +103,10 @@
   revision = "c8a8c6429d10e3b6865960ad8cb43779b8a834ef"
 
 [[projects]]
-  branch = "master"
   name = "github.com/juju/ratelimit"
   packages = ["."]
   revision = "59fac5042749a5afb9af70e813da1dd5474f0167"
+  version = "1.0.1"
 
 [[projects]]
   branch = "master"
@@ -136,13 +130,13 @@
   branch = "master"
   name = "github.com/koding/websocketproxy"
   packages = ["."]
-  revision = "424c31d1c6ac4384ddd4aaa81608310a98c3fc95"
+  revision = "944ae4ae170f3afa1e0683bcc3230d34c21e70eb"
 
 [[projects]]
   branch = "master"
   name = "github.com/lann/builder"
   packages = ["."]
-  revision = "f22ce00fd9394014049dad11c244859432bd6820"
+  revision = "1b87b36280d04fe7882d1512bf038ea2967ad534"
 
 [[projects]]
   branch = "master"
@@ -159,11 +153,8 @@
 [[projects]]
   branch = "master"
   name = "github.com/lib/pq"
-  packages = [
-    ".",
-    "oid"
-  ]
-  revision = "83612a56d3dd153a94a629cd64925371c9adad78"
+  packages = [".","oid"]
+  revision = "90697d60dd844d5ef6ff15135d0203f65d2f53b8"
 
 [[projects]]
   name = "github.com/mattn/go-colorable"
@@ -181,73 +172,59 @@
   branch = "master"
   name = "github.com/mitchellh/cli"
   packages = ["."]
-  revision = "33edc47170b5df54d2588696d590c5e20ee583fe"
+  revision = "c48282d14eba4b0817ddef3f832ff8d13851aefd"
 
 [[projects]]
   name = "github.com/posener/complete"
-  packages = [
-    ".",
-    "cmd",
-    "cmd/install",
-    "match"
-  ]
-  revision = "dc2bc5a81accba8782bebea28628224643a8286a"
-  version = "v1.1"
-
-[[projects]]
-  name = "github.com/satori/go.uuid"
-  packages = ["."]
-  revision = "879c5887cd475cd7864858769793b2ceb0d44feb"
-  version = "v1.1.0"
+  packages = [".","cmd","cmd/install","match"]
+  revision = "98eb9847f27ba2008d380a32c98be474dea55bdf"
+  version = "v1.1.1"
 
 [[projects]]
   branch = "master"
+  name = "github.com/satori/go.uuid"
+  packages = ["."]
+  revision = "36e9d2ebbde5e3f13ab2e25625fd453271d6522e"
+
+[[projects]]
   name = "github.com/ugorji/go"
   packages = ["codec"]
-  revision = "ccfe18359b55b97855cee1d3f74e5efbda4869dc"
+  revision = "b4c50a2b199d93b13dc15e78929cfb23bfdf21ab"
+  version = "v1.1.1"
 
 [[projects]]
   branch = "master"
   name = "golang.org/x/crypto"
   packages = ["ssh/terminal"]
-  revision = "0fcca4842a8d74bfddc2c96a073bd2a4d2a7a2e8"
+  revision = "027cca12c2d63e3d62b670d901e8a2c95854feec"
 
 [[projects]]
   branch = "master"
   name = "golang.org/x/net"
   packages = ["context"]
-  revision = "42fe2e1c20de1054d3d30f82cc9fb5b41e2e3767"
+  revision = "db08ff08e8622530d9ed3a0e8ac279f6d4c02196"
 
 [[projects]]
   branch = "master"
   name = "golang.org/x/sys"
-  packages = [
-    "unix",
-    "windows"
-  ]
-  revision = "a3f2cbd54cf5dfe3fbaccf76375fdb12f67654c8"
+  packages = ["unix","windows"]
+  revision = "6c888cc515d3ed83fc103cf1d84468aad274b0a7"
 
 [[projects]]
   branch = "v2"
   name = "gopkg.in/mgo.v2"
-  packages = [
-    ".",
-    "bson",
-    "internal/json",
-    "internal/sasl",
-    "internal/scram"
-  ]
+  packages = [".","bson","internal/json","internal/sasl","internal/scram"]
   revision = "3f83fa5005286a7fe593b055f0d7771a7dce4655"
 
 [[projects]]
-  branch = "v2"
   name = "gopkg.in/yaml.v2"
   packages = ["."]
-  revision = "0e4404da71227dcc02fb1deee803d93e86d08f72"
+  revision = "5420a8b6744d3b0345ab293f6fcba19c978f1183"
+  version = "v2.2.1"
 
 [solve-meta]
   analyzer-name = "dep"
   analyzer-version = 1
-  inputs-digest = "afe5e9587bf3c94201eed941f85330d5a9822fc9a4d43a672efe01cae9e7654e"
+  inputs-digest = "c20f9cb7fa5f208c2f79bc997d789636f027f044dfe9f724582763673f7070b4"
   solver-name = "gps-cdcl"
   solver-version = 1

--- a/Gopkg.toml
+++ b/Gopkg.toml
@@ -85,8 +85,8 @@
   name = "github.com/mitchellh/cli"
 
 [[constraint]]
+  branch = "master"
   name = "github.com/satori/go.uuid"
-  version = "1.1.0"
 
 [[constraint]]
   branch = "master"

--- a/kite.go
+++ b/kite.go
@@ -171,7 +171,7 @@ func NewWithConfig(name, version string, cfg *config.Config) *Kite {
 		panic("kite: version must be 3-digits semantic version")
 	}
 
-	kiteID := uuid.NewV4()
+	kiteID := uuid.Must(uuid.NewV4())
 
 	l, setlevel := newLogger(name)
 

--- a/kite_test.go
+++ b/kite_test.go
@@ -96,7 +96,7 @@ func TestContext(t *testing.T) {
 	}
 
 	if !reflect.DeepEqual(got, want) {
-		t.Fatalf("got %v, want %v")
+		t.Fatalf("got %v, want %v", got, want)
 	}
 }
 

--- a/kitetest/kitetest.go
+++ b/kitetest/kitetest.go
@@ -36,7 +36,7 @@ func (k *KiteKey) id() string {
 	if k.ID != "" {
 		return k.ID
 	}
-	return uuid.NewV4().String()
+	return uuid.Must(uuid.NewV4()).String()
 }
 
 func (k *KiteKey) issuer() string {

--- a/kontrol/bench_test.go
+++ b/kontrol/bench_test.go
@@ -12,7 +12,7 @@ func BenchmarkPostgres(b *testing.B) {
 	kon.SetStorage(NewPostgres(nil, kon.Kite.Log))
 
 	newKite := func() *protocol.Kite {
-		id := uuid.NewV4()
+		id := uuid.Must(uuid.NewV4())
 		return &protocol.Kite{
 			Username:    "bench-user",
 			Environment: "bench-env",
@@ -53,7 +53,7 @@ func BenchmarkEtcdAdd(b *testing.B) {
 	kon.SetStorage(NewEtcd(nil, kon.Kite.Log))
 
 	newKite := func() *protocol.Kite {
-		id := uuid.NewV4()
+		id := uuid.Must(uuid.NewV4())
 		return &protocol.Kite{
 			Username:    "bench-user",
 			Environment: "bench-env",

--- a/kontrol/kontrol.go
+++ b/kontrol/kontrol.go
@@ -283,7 +283,10 @@ func (k *Kontrol) AddKeyPair(id, public, private string) error {
 	}
 
 	if id == "" {
-		i := uuid.NewV4()
+		i, err := uuid.NewV4()
+		if err != nil {
+			return err
+		}
 		id = i.String()
 	}
 
@@ -358,12 +361,16 @@ func (k *Kontrol) InitializeSelf() error {
 }
 
 func (k *Kontrol) registerUser(username, publicKey, privateKey string) (kiteKey string, err error) {
+	id, err := uuid.NewV4()
+	if err != nil {
+		return "", err
+	}
 	claims := &kitekey.KiteClaims{
 		StandardClaims: jwt.StandardClaims{
 			Issuer:   k.Kite.Kite().Username,
 			Subject:  username,
 			IssuedAt: time.Now().Add(-k.tokenLeeway()).UTC().Unix(),
-			Id:       uuid.NewV4().String(),
+			Id:       id.String(),
 		},
 		KontrolURL: k.Kite.Config.KontrolURL,
 		KontrolKey: strings.TrimSpace(publicKey),
@@ -400,12 +407,17 @@ func (k *Kontrol) registerSelf() {
 		// to generate its kitekey or no kitekey is defined,
 		// use a dummy entry in order to register the kontrol.
 		keyPair = &KeyPair{
-			ID:      uuid.NewV4().String(),
 			Public:  "kontrol-self",
 			Private: "kontrol-self",
 		}
 
-		if err := k.keyPair.AddKey(keyPair); err != nil {
+		if id, err := uuid.NewV4(); err == nil {
+			keyPair.ID = id.String()
+
+			if err := k.keyPair.AddKey(keyPair); err != nil {
+				k.log.Error("%s", err)
+			}
+		} else {
 			k.log.Error("%s", err)
 		}
 	}
@@ -562,6 +574,11 @@ func (k *Kontrol) generateToken(tok *token) (string, error) {
 		return "", err
 	}
 
+	id, err := uuid.NewV4()
+	if err != nil {
+		return "", err
+	}
+
 	now := time.Now().UTC()
 
 	claims := &kitekey.KiteClaims{
@@ -571,7 +588,7 @@ func (k *Kontrol) generateToken(tok *token) (string, error) {
 			Audience:  tok.audience,
 			ExpiresAt: now.Add(k.tokenTTL()).Add(k.tokenLeeway()).UTC().Unix(),
 			IssuedAt:  now.Add(-k.tokenLeeway()).UTC().Unix(),
-			Id:        uuid.NewV4().String(),
+			Id:        id.String(),
 		},
 	}
 

--- a/kontrol/kontrol_test.go
+++ b/kontrol/kontrol_test.go
@@ -36,7 +36,7 @@ func init() {
 
 func TestUpdateKeys(t *testing.T) {
 	if storage := os.Getenv("KONTROL_STORAGE"); storage != "postgres" {
-		t.Skip("skipping TestUpdateKeys for storage %q: not implemented", storage)
+		t.Skipf("skipping TestUpdateKeys for storage %q: not implemented", storage)
 	}
 
 	kon, conf := startKontrol(testkeys.PrivateThird, testkeys.PublicThird, 5501)
@@ -661,10 +661,10 @@ func TestGetQueryKey(t *testing.T) {
 
 func TestKontrolMultiKey(t *testing.T) {
 	if storage := os.Getenv("KONTROL_STORAGE"); storage != "postgres" {
-		t.Skip("%q storage does not currently implement soft key pair deletes", storage)
+		t.Skipf("%q storage does not currently implement soft key pair deletes", storage)
 	}
 
-	i := uuid.NewV4()
+	i := uuid.Must(uuid.NewV4())
 	secondID := i.String()
 
 	// add so we can use it as key

--- a/testutil/testutil.go
+++ b/testutil/testutil.go
@@ -34,7 +34,7 @@ func NewKiteKeyWithKeyPair(private, public string) *jwt.Token {
 // NewToken creates new JWT token for the gien username. It embedds the given
 // public key as kontrolKey and signs the token with the private one.
 func NewToken(username, private, public string) *jwt.Token {
-	tknID := uuid.NewV4()
+	tknID := uuid.Must(uuid.NewV4())
 
 	hostname, err := os.Hostname()
 	if err != nil {

